packages/client/screens/TransactionsListScreen.tsx

```diff
@@ -4,17 +4,74 @@ import { CompositeScreenProps } from "@react-navigation/native";
 import { HomeStackScreenProps, SettingsStackScreenProps } from "../types";
 import { View } from "../components/Themed";
 import { TransactionsList } from "../components/transactions-list/TransactionsList";
+import { Pressable, Text, HStack, ScrollView } from "native-base";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { farBarsFilter } from "../components/icons/regular/farBarsFilter";
+import { facCircleXMark } from "../components/icons/solid/fasCircleXMark";
+import { getCurrencyType } from "@cryptify/common/src/utils/currency_utils";
+import { filterTransction } from "../services/filter_service";
 
 export default function TransactionsListScreen(
     props: CompositeScreenProps<
         HomeStackScreenProps<"TransactionsListScreen">,
         SettingsStackScreenProps<"TransactionsListScreen">
     >,
 ) {
+    const [filters, setFilters] = React.useState<string[]>([]);
+
+    React.useEffect(() => {
+        (() => {
+            props.navigation.setOptions({
+                headerRight: () => (
+                    <Pressable
+                        onPress={() => {
+                            props.navigation.navigate("FilterScreen", {
+                                setFilters,
+                                walletAddress: props.route.params.walletAddress,
+                            });
+                        }}
+                    >
+                        <FontAwesomeIcon icon={farBarsFilter} size={20} />
+                    </Pressable>
+                ),
+            });
+        })();
+    }, []);
+
+    const filtersDisplayed = filters.filter((f) => f !== "All transactions");
+
+    const transactions = props.route.params.transactions;
+    const walletAddress = props.route.params.walletAddress;
+    const type = getCurrencyType(walletAddress);
+
+    const DisplayedTransaction = filterTransction(type, walletAddress, transactions, filtersDisplayed);
+
+    function FiltersBadges() {
+        return (
+            <View style={{ height: 50 }}>
+                <ScrollView horizontal>
+                    {filtersDisplayed.map((filter) => (
+                        <HStack key={filter} style={styles.badge}>
+                            <Text style={styles.badgeText}>{filter}</Text>
```

Comment:
Add size, fontWeight and color directly as component properties

```suggestion
                            <Text size={"footnote1"} fontWeigh={"semibold"} color={"darkBlue.500"} style={styles.badgeText}>{filter}</Text>
```